lib/cloudant.js

```diff
@@ -524,21 +531,52 @@ Cloudant.prototype.updateAll = function(model, where, data, options, cb) {
           model, where, data, options);
   var self = this;
   var mo = self.selectModel(model);
+  var idName = self.idName(model);
+  // if idName exists, change the idName to be the default `_id` key
+  if (idName && data[idName]) {
+    data._id = data[idName];
+    delete data[idName];
+  }
+  var res = {};
+  var docList = [];
+
   self.all(model, {where: where}, {raw: true}, function(err, docs) {
     if (err) return cb(err, docs);
     if (docs.length === 0) return cb(null, {count: 0});
 
     data = getPlainJSONData.call(self, model, data);
+
+    /**
+     * for each of the docs from the given filter, only update the instance where the revs match
+     * behavior of bulk update in cloudant:
```

Comment:
I'm not sure what the uniqueness of `_rev` has to do with anything here. It is a per-document number with a digest appended to it so that you know that `2-deadbeef` is newer than `1-abcdef`, but you can't guess/forge what `3-??????` will be and accidentally overwrite it.